sdl_android/src/main/java/com/smartdevicelink/api/screen/TextAndGraphicManager.java

```diff
@@ -0,0 +1,893 @@
+						Log.v(TAG, "Queued image update matches the images we need, sending update");
+						sendShow();
+					} else {
+						Log.v(TAG, "Queued image update does not match the images we need, skipping update");
+					}
+				}
+			});
+			queuedImageUpdate = fullShow;
+		}
+	}
+
+	private void sendShow(){
+		inProgressUpdate.setOnRPCResponseListener(new OnRPCResponseListener() {
+			@Override
+			public void onResponse(int correlationId, RPCResponse response) {
+				if (response.getSuccess()){
+					Log.v(TAG, "Show Successful");
+					updateCurrentScreenDataFromShow(inProgressUpdate);
+				}
+
+				inProgressUpdate = null;
+				if (inProgressListener != null){
+					inProgressListener.onComplete(true);
+					inProgressListener = null;
+				}
+
+				if (hasQueuedUpdate){
+					Log.v(TAG, "Queued update exists, sending another update");
+					sdl_update(queuedUpdateListener);
+					queuedUpdateListener = null;
+					hasQueuedUpdate = false;
+				}
+			}
+		});
+
+		this.softButtonManager.setCurrentMainField1(inProgressUpdate.getMainField1());
+		internalInterface.sendRPCRequest(inProgressUpdate);
+	}
+
+	// Images
+
+	private void uploadImages(final CompletionListener listener) {
+
+		List<SdlArtwork> artworksToUpload = new ArrayList<>();
+
+		// add primary image
+		if (shouldUpdatePrimaryImage()){
+			artworksToUpload.add(primaryGraphic);
+		}
+
+		// add secondary image
+		if (shouldUpdateSecondaryImage()){
+			artworksToUpload.add(secondaryGraphic);
+		}
+		Log.i(TAG, "Artworks to upload: "+ artworksToUpload.toString());
+		// use file manager to upload art
+		fileManager.uploadArtworks(artworksToUpload, new MultipleFileCompletionListener() {
+			@Override
+			public void onComplete(Map<String, String> errors) {
+				if (errors != null) {
+					Log.e(TAG, "Error Uploading Artworks. Error: " + errors.toString());
+					listener.onComplete(false);
+				}else{
+					Log.v(TAG, "Successfully uploaded Artworks");
+					listener.onComplete(true);
+				}
+			}
+		});
+	}
+
+	private Show assembleShowImages(Show show){
+
+		if (!shouldUpdatePrimaryImage() && ! shouldUpdateSecondaryImage()){
+			return show;
+		}
+
+		if (shouldUpdatePrimaryImage()){
+			Image primaryImage = new Image();
+			primaryImage.setImageType(ImageType.DYNAMIC);
+			primaryImage.setValue(primaryGraphic.getName());
+			show.setGraphic(primaryImage);
+		}
+
+		if (shouldUpdateSecondaryImage()){
+			Image secondaryImage = new Image();
+			secondaryImage.setImageType(ImageType.DYNAMIC);
+			secondaryImage.setValue(secondaryGraphic.getName());
+			show.setSecondaryGraphic(secondaryImage);
+		}
+
+		return show;
+	}
+
+	// Text
+
+	protected Show assembleShowText(Show show){
+
+		show = setBlankTextFields(show);
+
+		if (mediaTrackTextField != null){
+			show.setMediaTrack(mediaTrackTextField);
+		} else {
+			show.setMediaTrack("");
```

Comment:
This is unnecessary as the previous method call will sets this field to empty set